public/app/core/components/AppChrome/MegaMenu/utils.ts

```diff
@@ -35,21 +35,36 @@ export const enrichHelpItem = (helpItem: NavModelItem) => {
   return helpItem;
 };
 
-export const enrichWithInteractionTracking = (item: NavModelItem, megaMenuDockedState: boolean) => {
+export const enrichWithInteractionTracking = (
+  item: NavModelItem,
+  megaMenuDockedState: boolean,
+  ancestorIsNew = false
+) => {
   // creating a new object here to not mutate the original item object
   const newItem = { ...item };
   const onClick = newItem.onClick;
+
+  let isNew: 'item' | 'ancestor' | undefined = undefined;
+  if (newItem.isNew) {
+    isNew = 'item';
+  } else if (ancestorIsNew) {
+    isNew = 'ancestor';
+  }
+
   newItem.onClick = () => {
     reportInteraction('grafana_navigation_item_clicked', {
       path: newItem.url ?? newItem.id,
       menuIsDocked: megaMenuDockedState,
       itemIsBookmarked: Boolean(config.featureToggles.pinNavItems && newItem?.parentItem?.id === 'bookmarks'),
       bookmarkToggleOn: Boolean(config.featureToggles.pinNavItems),
+      isNew,
```

Comment:
I might not be getting this but, here is`isNew` a 'string | undefined'  and in line 66 a boolean?

Reply:
yup - we send through `'item' | 'ancestor' | undefined` to rudderstack.

When calling `enrichWithInteractionTracking` to set the tracking info for child items, we only need to pass through a boolean of whether one of the parents/ancestors isNew.